### DISCUSSION TITLE
Side by side install of eBPF

### DISCUSSION THREAD

**Discussion by shankarseal** [microsoft/ebpf-for-windows#1916] [General]
It can be envisioned that the eBPF core components would be deployed by the product that takes a dependency on them. Lets consider two products Prod1 and Prod2 that takes a dependency on version N and version M of eBPF core respectively. What happens on a system which has both the products installed? Do we support SxS installation of eBPF core components? Or is the latest eBPF core that gets precedence? That would mean eBPF core must be backward compatible. If we choose MSI as the installation technology for eBPF; then the MSI must have logic to ensure that a version eBPF can never supersede an existing installation with higher version.

**Comment by saxena-anurag** [comment]
If eBPF-for-windows always guarantees backward compatibility, then I think we don't need to support SxS?

**Reply by dthaler** [reply]
Agreed, we should try to guarantee backwards compatibility within major versions.  I believe this goes along with our existing statement that we adhere to [semver 2.0](https://semver.org/).   And we should try not to rev the major version if we can avoid it.  That means we use v1.0 when we go to production, and can then use 1.x.x without SxS and only have to deal with SxS if we ever need to go to a non-backward compatible 2.0
